extensions/functions_string.yaml

```diff
@@ -1263,6 +1263,74 @@ scalar_functions:
           - value: i32
             name: "count"
         return: "string"
+  -
+    name: string_split
+    description: >-
+      Split a string into a list of strings, based on a specified `separator` character.
+    impls:
+      - args:
+          - value: "varchar<L1>"
+            name: "input"
+            description: The input string.
+          - value: "varchar<L2>"
+            name: "separator"
+            description: A character used for splitting the string.
+        return: "List<varchar<L1>>"
```

Comment:
Wasn't exactly sure how to put the return type.

Reply:
Looks good to me.